app/views/cart/show.html.erb

```diff
@@ -1,5 +1,7 @@
 <h1>Cart</h1>
-
+<% if !current_user %>
+  <p>To checkout you must <%= link_to 'log in', '/login' %> or <%= link_to 'register', '/register' %></p>
```

Comment:
this message will persist to encourage the visitor to log in or register before checking out

Reply:
i like this feature